clang/lib/Headers/hlsl/hlsl_intrinsics.h

```diff
@@ -2015,6 +2015,13 @@ _HLSL_AVAILABILITY(shadermodel, 6.0)
 _HLSL_BUILTIN_ALIAS(__builtin_hlsl_wave_is_first_lane)
 __attribute__((convergent)) bool WaveIsFirstLane();
 
+// \brief Returns the value of the expression for the given lane index within
+// the specified wave.
+template <typename T>
+_HLSL_AVAILABILITY(shadermodel, 6.0)
+_HLSL_BUILTIN_ALIAS(__builtin_hlsl_wave_read_lane_at)
+    __attribute__((convergent)) T WaveReadLaneAt(T, int32_t);
```

Comment:
Despite the documentation, this doesn't seem to work in DXC for arbitrary types:

https://godbolt.org/z/48brf1bre

@tex3d, gen_intrin_main says this intrinsic is valid on `any<>`, which seems to be any vector or scalar. Is that correct?

Reply:
Fwiw, this is also true for the `select` intrinsic and we use the arbitrary type template.

Reply:
`any<>` means scalar, vector, or matrix, with any component type valid with those (bool and basic numeric types).

Reply:
> Fwiw, this is also true for the `select` intrinsic and we use the arbitrary type template.

`select` is a little different, there are two overloads of it, one for numeric types and one for samplers. Really it should have worked with any type, but gen_intrin_main isn't expressive enough.